client2/timer_queue.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: © 2023 David Stainton
+// SPDX-License-Identifier: AGPL-3.0-only
+package client2
+
+import (
+	"sync"
+	"time"
+
+	"github.com/katzenpost/katzenpost/core/queue"
+	sConstants "github.com/katzenpost/katzenpost/core/sphinx/constants"
+	"github.com/katzenpost/katzenpost/core/worker"
+)
+
+type TimerQueue struct {
+	worker.Worker
+
+	queue  *queue.PriorityQueue
+	timer  *time.Timer
+	mutex  sync.Mutex
```

Comment:
maybe this could be an RWLock to enable non-blocking `t.Len()`  and `t.queue.Peek()` ?

Reply:
fixed